src/script/src/python/utils.rs

```diff
@@ -113,3 +114,16 @@ pub fn py_vec_obj_to_array(
         ret_other_error_with(format!("Expect a vector or a constant, found {obj:?}")).fail()
     }
 }
+
+/// a terrible hack to call async from sync by:
+/// TODO(discord9): find a better way
+/// 1. spawn a new thread
+/// 2. create a new runtime in new thread and call `block_on` on it
+pub fn block_on_async<T, F>(f: F) -> std::thread::Result<T>
```

Comment:
Can we use our `Runtime::block_on`?

Reply:
> Can we use our `Runtime::block_on`?

It seems I can only acquire a `Handle` for blocking, and handle can cause blocking because only the Runtime::block_on method can drive the IO and timer drivers, but the Handle::block_on method cannot drive them.(I encounter this blocking caused by handle::block_on in #884 ) here is the doc link about it: https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#method.block_on